src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,57 +1,160 @@
-import React from 'react';
+import { FC, memo, useState } from 'react';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieData } from '../../types/MovieData';
+import { MovieCard } from '../MovieCard';
 import './FindMovie.scss';
 
-export const FindMovie: React.FC = () => {
-  return (
-    <>
-      <form className="find-movie">
-        <div className="field">
-          <label className="label" htmlFor="movie-title">
-            Movie title
-          </label>
-
-          <div className="control">
-            <input
-              data-cy="titleField"
-              type="text"
-              id="movie-title"
-              placeholder="Enter a title to search"
-              className="input is-dander"
-            />
-          </div>
+interface Props {
+  onAdd: React.Dispatch<React.SetStateAction<Movie[]>>,
+}
+
+export const FindMovie: FC<Props> = memo(
+  ({ onAdd }) => {
+    const [title, setTitle] = useState('');
+    const [movie, setMovie] = useState<Movie | null>(null);
+    const [isLoading, setIsLoading] = useState(false);
+    const [error, setError] = useState(false);
+
+    const handleGetMovie = (
+      event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+    ) => {
+      event.preventDefault();
+      setIsLoading(true);
+
+      getMovie(title)
+        .then(response => {
+          if ('Error' in response) {
+            throw new Error('error');
+          }
+
+          const {
+            Poster,
+            Title,
+            Plot,
+            imdbID,
+          } = response as MovieData;
+
+          const posterForMovie = Poster === 'N/A'
+            ? 'https://via.placeholder.com/360x270.png?text=no%20preview'
+            : Poster;
+
+          setMovie({
+            title: Title,
+            description: Plot,
+            imgUrl: posterForMovie,
+            imdbUrl: `https://www.imdb.com/title/${imdbID}`,
+            imdbId: imdbID,
+          });
+        })
+        .catch(() => {
+          setError(true);
+        })
+        .finally(() => setIsLoading(false));
+    };
+
+    const handleMovieAdding = () => {
+      onAdd((movies) => {
+        if (!movie) {
+          return movies;
+        }
+
+        const movieToAdd = {
+          ...movie,
+        };
+
```

Comment:
let's move this method to parent component and pass it here as props

and do not forget to display an error when the film is already in the list